strawberryfields/engine.py

```diff
@@ -367,21 +367,34 @@ def run(self, program, *, compile_options=None, run_options=None):
                 and state. TF backend only.
         """
         compile_options = compile_options or {}
-        run_options = run_options or {}
-        run_options.setdefault("shots", 1)
-        run_options.setdefault('modes', None)
+        temp_run_options = {}
+
+        if isinstance(program, Sequence):
+            # succesively update all run option defaults.
```

Comment:
Feels like this is something that warrants a dedicated test

Reply:
Will add.

Reply:
Added in 8fe10dd2eb61bfb02b4c20942f347e8aa3493ca6